docs/DeveloperGuide.md

```diff
@@ -1,34 +1,503 @@
+   2. [Run the tests](TestingGuide.md) to ensure they all pass. 
+
+## 3.0 Design
+
+This section explains the high-level design of the application. Given below is an overview of each component and a more detailed explanation of the architecture. Trippie is the main class which manages the initialization of the relevant classes and their execution.
+
+
+### 3.1 Architecture (Felix)
+
+<kbd>
+
+![UML Diagram Main](https://i.imgur.com/V3vUpoo.jpg)
+<center><i>Figure 1: Overall Architecture</i></center>
+
+</kbd>
+
+The Architecture Diagram given above explains the high-level design of Trippie. Shown below is a diagram of Trippie's high-level structure.
+
+The main class of`Trippie` is called [`Trippie.java`](https://github.com/AY2021S1-CS2113T-W11-2/tp/blob/master/src/main/java/seedu/trippie/Trippie.java?) It is responsible for:
+* At app launch: Initializes the components in the correct sequence, and connects them up with each other.
+* At shut down: Shuts down the components and invokes cleanup methods where necessary.
+
+The rest of the App consists of three components.
+- **UI** : The UI of the App.
+- **Parser** : Parses user input and its information to respective commands.
+- **Storage** : Reads data from, and write data to, the text file.
+- **Command** : Executes each command respectively.
+
+The Sequence diagram below shows how the program flows across the different classes when parsing or executing user commands.
+
+![UML Sequence Diagram](https://i.imgur.com/hHq7ltY.png)
+<center><i>Figure 2: Sequence Diagram during Command execution</i></center>
+
+### 3.2 Ui (Felix)
+
+The UI class is in charge of the user's input, from the `readCommand` or `getLine` methods. Other functions of the Ui is to provide shorthands of Trippie's outputs, such as `greetUser` or `showFarewells`.
+
+### 3.3 Parser (Wei Shuang)
+The `Parser` class is implemented to parse the user's input, and returns a `Command` object. The `Trippie` class will read in this object and call the `execute` method under the object.
+
+When the `Parser` class is unable to successfully parse the user's input, `TrippieIllegalCommandException` will be thrown. (Refer to 3.7 Exceptions)
+
+### 3.4 TrippieData (Ivander)
+
+The `TrippieData` class is implemented to store all of your data during Trippie's runtime. Below is an architecture diagram of the class.
+
+<kbd>
+
+![TrippieData UML Diagram](https://i.imgur.com/3eAMiob.jpg)
+<center><i>Figure 3: Class Architecture</i></center>
+
+</kbd>
+
+The explanation for the above *Figure 3* is as follows:
+1. `TrippieData` can contain many `Trip` objects.
+2. Each `Trip` consists of a `PlaceList` and an `ExpenseList`.
+3. `PlaceList` contains an ArrayList of `Places`.
+4. `ExpenseList` contains an ArrayList of `Expense`.
+
+To get the name of a `Place` of index 5 in `PlaceList` and from Trip of index 4 in TrippieData, the program will be implemented based on the following UML diagram.
```

Comment:
The diagram seems to be getting Trip of index 5 and Place of index 4 instead, could this be a typo in the diagram?